### ISSUE TITLE
@rollup/plugin-node-resolve: Unresolved import due to nested sub-module paths in exports field

### ISSUE THREAD

**Issue by carsonfarmer** [rollup/plugins#641]
Firstly, thanks for the amazing work on the @rollup/plugin-node-resolve plugin, its a life-saver. Especially happy to have the latest work on supporting package entry points! This issue relates to how the plugin handles nested paths in package entry points. In theory, the examples in the repl.it should be allowed: https://nodejs.org/api/packages.html#packages_package_entry_points

- Rollup Plugin Name: @rollup/plugin-node-resolve
- Rollup Plugin Version:10.0.0
- Rollup Version:2.33.1
- Operating System (or Browser): MacOS Catalina (10.15.6)
- Node Version: v13.10.1
- Link to reproduction _(⚠️ read below)_: https://repl.it/@carsonfarmer/node-resolve-unresolved-deps

### Expected Behavior

To be able to resolve packages with export fields that contain nested paths. This seems like a common scenario:

```
...
"exports": {
    ".": {
      "require": "./cjs/src/index.js",
      "import": "./esm/src/index.js",
      "browser": "./esm/src/browser.js"
    }
},
...
```

or 

```
...
"exports": {
    ".": "./foo/nested/c.js",
    "./bar": "./bar/a.js"
  }
...
```

or any of the examples in https://nodejs.org/api/packages.html#packages_package_entry_points

### Actual Behavior

The plugin is unable to resolve nested paths. Please see attached repl.it for a minimal reproducible example of this.

**Comment by lukastaegert** [comment]
Well, it has not been released yet. Please file any issues you find after it the feature has actually been released: https://github.com/rollup/plugins/blob/master/packages/node-resolve/CHANGELOG.md

**Comment by lukastaegert** [comment]
Note that the feature has already been merged and will be part of the next release: #540

**Comment by carsonfarmer** [comment]
Apologies for the noise, I don't know how I missed that: 🤦. Looking forward to the release!

**Comment by aomarks** [comment]
> Note that the feature has already been merged and will be part of the next release: #540

Any ETA on release? (No pressure, just curious, eagerly awaiting the new package entry points feature!)
